File: src/Services/ArchT.Services.Inventory/Startup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ArchT.Services.Inventory.Infrastructure.Extensions;
using MediatR;
using Microsoft.AspNetCore.Authentication.JwtBearer;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;
using Microsoft.Extensions.Options;
using Microsoft.IdentityModel.Tokens;
using Swashbuckle.AspNetCore.Swagger;

namespace ArchT.Services.Inventory
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
            services.AddSwaggerGen(swagger =>
            {
                swagger.SwaggerDoc("v1", new Info { Title = "Inventory Service API", Version = "v1" });
            });

            services.AddAuthentication().AddJwtBearer(options =>
                options.TokenValidationParameters = GenerateParams());

            services.AddAuthorization(auth =>
            {
                auth.AddPolicy("JwtPolicy", new AuthorizationPolicyBuilder()
                    .AddAuthenticationSchemes(JwtBearerDefaults.AuthenticationScheme)
                    .RequireAuthenticatedUser()
                    .Build());
            });

            services.AddCors(options =>
            {
                options.AddPolicy("CorsPolicy",
                    builder => builder.AllowAnyOrigin()
                        .AllowAnyMethod()
                        .AllowAnyHeader()
                        .AllowCredentials());
            });

            services.AddInfrastructureServices();
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            app.UseMvc()
                .UseAuthentication()
                .UseCors("CorsPolicy")
                .UseSwagger()
                .UseSwaggerUI(options =>
                {
                    options.SwaggerEndpoint($"/swagger/v1/swagger.json", $"v1");
                    options.RoutePrefix = string.Empty;
                    options.OAuthClientId("client-id");
                    options.OAuthClientSecret(Convert.ToBase64String(Encoding.UTF8.GetBytes("client-top-secret")));
                    options.OAuthRealm("");
                    options.OAuthAppName("");
                });
        }

        TokenValidationParameters GenerateParams()
        {
            var secret = Convert.ToBase64String(Encoding.UTF8.GetBytes("client-top-secret"));
            var key = Encoding.ASCII.GetBytes(secret);
            var credentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature);

            return new TokenValidationParameters
            {
                IssuerSigningKey = credentials.Key,
                ValidAudience = "Audience",
                ValidIssuer = "Issuer",
                ValidateIssuerSigningKey = true,
                ValidateLifetime = true,
                ClockSkew = TimeSpan.Zero
            };
        }
    }
}

File: src/Services/ArchT.Services.Trade/Startup.fs
namespace ArchT.Services.Trade

open System
open System.Collections.Generic
open System.Linq
open System.Threading.Tasks
open Microsoft.AspNetCore.Builder
open Microsoft.AspNetCore.Hosting
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Configuration
open Microsoft.Extensions.DependencyInjection
open Swashbuckle.AspNetCore.Swagger
open Configurations

type Startup private () =
    new (configuration: IConfiguration) as this =
        Startup() then
        this.Configuration <- configuration

    // This method gets called by the runtime. Use this method to add services to the container.
    member this.ConfigureServices(services: IServiceCollection) =
        // Add framework services.
        services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1) |> ignore

        services.AddSwaggerGen(fun swagger ->
            let info = Info()
            info.Title <- "Trade Service API"; info.Version <- "v1"
            swagger.SwaggerDoc("v1", info)) |> ignore

        services.AddCors(fun opt ->
            opt.AddPolicy("CorsPolicy", fun b -> b.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader().AllowCredentials() |> ignore)
        ) |> ignore

    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
    member this.Configure(app: IApplicationBuilder, env: IHostingEnvironment) =
        if (env.IsDevelopment()) then
            app.UseDeveloperExceptionPage() |> ignore

        app.UseMvc()
            .UseCors("CorsPolicy")
            .UseSwagger()
            .UseSwaggerUI(fun opt ->
                opt.SwaggerEndpoint("/swagger/v1/swagger.json", "v1")
                opt.RoutePrefix <- ""
             ) |> ignore

    member val Configuration : IConfiguration = null with get, set